build-docbook-catalog

```diff
@@ -137,8 +137,10 @@ set_dtds() {
 	local d=${ROOT}${DOCBOOKDIR}
 	if [[ -d ${d} ]] ; then
 		pushd "${d}" >/dev/null || return 1
-		mapfile -d $'\0' DTDS < <(find xml-dtd-*/ -name docbookx.dtd -print0)
-		mapfile -d $'\0' SIMPLE_DTDS < <(find xml-simple-dtd-*/ -name sdocbook.dtd -print0)
+		mapfile -d $'\0' DTDS < <(
+			find -path "./xml-dtd-*/docbookx.dtd" -printf "%P\0")
```

Comment:
`-print0` is in POSIX.  `-printf` is not.

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/find.html

since this script is bash, if we're going to assume that the dtd only ever exists in subdirs with depth=1, why not leverage nullglob ?

```
shopt -s nullglob
DTDS=( xml-dtd-*/docbookx.dtd )
SIMPLE_DTDS=( xml-simple-dtd-*/sdocbook.dtd )
shopt -u nullglob
```

Reply:
SGTM